engine/swarm/configs.md

```diff
@@ -0,0 +1,569 @@
+
+## Read more about `docker config` commands
+
+Use these links to read about specific commands, or continue to the
+[example about using configs with a service](configs.md#example-use-configs-with-a-service).
+
+- [`docker config create`](/engine/reference/commandline/config_create.md)
+- [`docker config inspect`](/engine/reference/commandline/config_inspect.md)
+- [`docker config ls`](/engine/reference/commandline/config_ls.md)
+- [`docker config rm`](/engine/reference/commandline/config_rm.md)
+
+## Examples
+
+This section includes graduated examples which illustrate how to use
+Docker configs.
+
+> **Note**: These examples use a single-Engine swarm and unscaled services for
+> simplicity. The examples use Linux containers, but Windows containers also
+> support configs.
+
+### Simple example: Get started with configs
+
+This simple example shows how configs work in just a few commands. For a
+real-world example, continue to
+[Intermediate example: Use configs with a Nginx service](#advanced-example-use-configs-with-a-nginx-service).
+
+1.  Add a config to Docker. The `docker config create` command reads standard
+    input because the last argument, which represents the file to read the
+    config from, is set to `-`.
+
+    ```bash
+    $ echo "This is a config" | docker config create my-config -
+    ```
+
+2.  Create a `redis` service and grant it access to the config. By default,
+    the container can access the config at `/my-config`, but
+    you can customize the file name on the container using the `target` option.
+
+    ```bash
+    $ docker service  create --name redis --config my-config redis:alpine
+    ```
+
+3.  Verify that the task is running without issues using `docker service ps`. If
+    everything is working, the output looks similar to this:
+
+    ```bash
+    $ docker service ps redis
+
+    ID            NAME     IMAGE         NODE              DESIRED STATE  CURRENT STATE          ERROR  PORTS
+    bkna6bpn8r1a  redis.1  redis:alpine  ip-172-31-46-109  Running        Running 8 seconds ago  
+    ```
+
+4.  Get the ID of the `redis` service task container using `docker ps`, so that
+    you can use `docker exec` to connect to the container and read the contents
+    of the config data file, which defaults to being readable by all and has the
+    same name as the name of the config. The first command below illustrates
+    how to find the container ID, and the second and third commands use shell
+    completion to do this automatically.
+
+    ```bash
+    $ docker ps --filter name=redis -q
+
+    5cb1c2348a59
+
+    $ docker exec $(docker ps --filter name=redis -q) ls -l /my-config
+
+    -r--r--r--    1 root     root            12 Jun  5 20:49 my-config                                                     
+
+    $ docker exec $(docker ps --filter name=redis -q) cat /my-config
+
+    This is a config
+    ```
+
+5.  Verify that the config is **not** available if you commit the container.
```

Comment:
I wonder if this is important information (in general, committing a container is not best practice, as the container should be regarded immutable). I would personally skip this.